crates/bevy_ecs/src/reflect.rs

```diff
@@ -62,6 +97,16 @@ impl ReflectComponent {
         (self.reflect_component_mut)(world, entity)
     }
 
+    /// Directly copies the value of the component of type `C` to a new entity
+    ///
+    /// This method creates a new component of type `C` using the [FromWorld] trait,
+    /// sets its value to the value of the component of type `C` on the `source_entity`,
+    /// and then inserts the new component into the `destination_entity`.
+    ///
+    /// WARNING: this method does not use the `Clone` method: instead, memory is copied directly.
```

Comment:
```suggestion
    /// **Note**: this method uses `Reflect` to create a shallow value-based copy of the component and will not respect `Clone` implementations.
```

Pointer types are already ignored/incompatible with Reflect as of now, so the ref-counting match really doesn't hold water here.